tools/c7n_azure/tests/azure_common.py

```diff
@@ -105,22 +130,103 @@ def request_callback(self, request):
             return None
         return request
 
+    def _response_callback(self, response):
+        if not hasattr(self, 'cassette') or os.path.isfile(self.cassette._path):
+            if 'data' in response['body']:
+                body = json.dumps(response['body']['data'])
+                response['body']['string'] = body.encode('utf-8')
+                response['headers']['content-length'] = [str(len(body))]
+
+            return response
+
+        headers = {}
+        for key in response['headers']:
+            if key.lower() not in self.FILTERED_HEADERS:
+                headers[key.lower()] = response['headers'][key]
+        response['headers'] = headers
+
+        content_type = response['headers'].get('content-type', (None,))[0]
+        if not content_type or 'application/json' not in content_type:
+            return response
+
+        body = response['body'].pop('string').decode('utf-8')
+
+        # Clean up subscription IDs in body
+        body = AzureVCRBaseTest._replace_subscription_id(body)
+
+        response['body']['data'] = json.loads(body)
```

Comment:
nit: I saw some cases when Azure returns `Content-Type: application/json` but it is not.. Not sure if you want to handle it :)

Reply:
Worried that automatically handling the exception and storing as a string will cover up bugs, as this json.loads fails constantly when things go wrong and it is a good sanity check.   If someone can't record due to legitimate API bug though they can enhance this - I haven't had it fail due to a bad response yet.

Reply:
seems like something to handle when we find it, ie. letting it blow up on test record seems appropriate, for human to come examine.

Reply:
I ended up finding it helpful to at least give an easier to track down test failure when it happens, so I am asserting something helpful here when we fail.